ctk/common/src/main/kotlin/org/codice/compliance/utils/TestCommon.kt

```diff
@@ -90,20 +99,60 @@ class TestCommon {
         const val PRIVATE_KEY_PASSWORD =
                 "org.apache.ws.security.crypto.merlin.keystore.private.password"
 
-        private val DEPLOY_CL = getDeployDirClassloader()
-        private val spMetadata = Common.parseSpMetadata()
+        private const val DEFAULT_SP_ISSUER = "https://samlhost:8993/services/saml"
+        private const val DSA_SP_ISSUER = "https://samlhostdsa:8993/services/samldsa"
+        @JvmField
+        var currentSPIssuer = DEFAULT_SP_ISSUER
 
-        val idpMetadata = parseAndVerifyVersion()
+        /*
+         * All of these properties are lazy, so that unit tests do not have to have all of this
+         * system-level information setup.
+         */
+        private val DEPLOY_CL by lazy {
+            getDeployDirClassloader()
+        }
+
+        val idpMetadata by lazy {
+            parseAndVerifyVersion()
+        }
+
+        private val spMetadata by lazy {
+            Common.parseSpMetadata()
+        }
+
+        private val DEFAULT_SP_ENTITY_INFO by lazy {
+            checkNotNull(spMetadata[DEFAULT_SP_ISSUER])
+        }
+
+        private val DSA_SP_ENTITY_INFO by lazy {
+            checkNotNull(spMetadata[DSA_SP_ISSUER])
+        }
+
+        var currentSPEntityInfo by LazyVar {
+            DEFAULT_SP_ENTITY_INFO
+        }
+
+        object UseDSASigningSP : TestCaseExtension {
+            override fun intercept(context: TestCaseInterceptContext,
+                                   test: (TestCaseConfig, (TestResult) -> Unit) -> Unit,
+                                   complete: (TestResult) -> Unit) {
+                currentSPIssuer = DSA_SP_ISSUER
+                currentSPEntityInfo = DSA_SP_ENTITY_INFO
 
-        val SP_ISSUER = spMetadata.keys.first()
-        val SP_ENTITY_INFO = checkNotNull(spMetadata[SP_ISSUER])
+                test(context.config, { complete(it) })
 
-        val acsUrl: Map<SamlProtocol.Binding, String?> by lazy {
-            val spInfo = SP_ENTITY_INFO
-            SamlProtocol.Binding.values()
-                    .associate {
-                        it to spInfo.getAssertionConsumerService(it)?.url
-                    }
+                currentSPIssuer = DEFAULT_SP_ISSUER
+                currentSPEntityInfo = DEFAULT_SP_ENTITY_INFO
+            }
+        }
+
+        @JvmStatic
```

Comment:
❓ Reading a bit about this. 
> Specifies that an additional static method needs to be generated from this element if it's a function.

Why is this necessary? Doesn't the fact that it's in a companion object make it static?